samples/client/petstore/php/OpenAPIClient-php/lib/ObjectSerializer.php

```diff
@@ -87,12 +87,16 @@ public static function sanitizeForSerialization($data, $type = null, $format = n
                 foreach ($data::openAPITypes() as $property => $openAPIType) {
                     $getter = $data::getters()[$property];
                     $value = $data->$getter();
-                    if ($value !== null
-                        && !in_array($openAPIType, ['DateTime', 'bool', 'boolean', 'byte', 'double', 'float', 'int', 'integer', 'mixed', 'number', 'object', 'string', 'void'], true)
-                        && method_exists($openAPIType, 'getAllowableEnumValues')
-                        && !in_array($value, $openAPIType::getAllowableEnumValues(), true)) {
-                        $imploded = implode("', '", $openAPIType::getAllowableEnumValues());
-                        throw new \InvalidArgumentException("Invalid value for enum '$openAPIType', must be one of: '$imploded'");
+                    if ($value !== null && !in_array($openAPIType, ['DateTime', 'bool', 'boolean', 'byte', 'double', 'float', 'int', 'integer', 'mixed', 'number', 'object', 'string', 'void'], true)) {
```

Comment:
@dkarlovi So... instead of one big condition you decided to use three nested conditions? It's weird that PHPStan doesn't complain about that. It usually says to me "consider refactoring" when I have the third nested condition. 😕

Reply:
@ybelenko the nested ifs are equivalent to the original single if, the difference is the safer types, in this case, `$callable` is really `callable`, using it instead of `method_exists`, see my other comment with links.

IMO the serializer should in the future likely be extracted from the package anyway since it's static and has tests which are outside of the generated package (hinting at its static nature). Refactoring it further until it's clean enough to split apart / replace is a good direction here.